sympy/solvers/decompogen.py

```diff
@@ -47,6 +50,14 @@ def decompogen(f, symbol):
         result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
         return result
 
+    # ===== Max/Min Functions ===== #
+    if isinstance(f, (Max, Min)):
+        result += [type(f)]
```

Comment:
Adding what you are going to traverse in the `for` loop makes this fragile -- trust recursion here:
```
if symbol not in f.free_symbols or f == symbol:
    return [f]

if isinstance(f, (Function, Pow)):
...
if isinstance(f, (Max, Min)):
    rv = [type(f)]
    for i in f.args:
        rv.extend(decompogen(i, symbol))
    return rv
```

But I don't know what is going to be done with the results and whether it is important to know that Min and Max were present...so maybe it should start out as `rv = []`.